src/renderer/components/+workloads-overview/detail-components.injectable.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable, lifecycleEnum } from "@ogre-tools/injectable";
+import { computed } from "mobx";
+import rendererExtensionsInjectable from "../../../extensions/renderer-extensions.injectable";
+import { OverviewStatuses } from "./overview-statuses";
+import { WorkloadEvents } from "../../initializers/workload-events";
+import { orderBy } from "lodash/fp";
+import type { WorkloadsOverviewDetailRegistration } from "./workloads-overview-detail-registration";
+
+const detailComponentsInjectable = getInjectable({
+  instantiate: (di) => {
+    const extensions = di.inject(rendererExtensionsInjectable);
+
+    return computed(() => {
+      const extensionRegistrations = extensions
+        .get()
+        .flatMap((extension) => extension.kubeWorkloadsOverviewItems);
+
+      const allRegistrations = [
+        ...coreRegistrations,
+        ...extensionRegistrations,
+      ];
+
+      return getRegistrationsInPriorityOrder(allRegistrations).map(
+        (item) => item.components.Details,
+      );
+    });
+  },
+
+  lifecycle: lifecycleEnum.singleton,
+});
+
+const coreRegistrations = [
+  {
+    components: {
+      Details: OverviewStatuses,
+    },
+  },
+  {
+    priority: 5,
+    components: {
+      Details: WorkloadEvents,
+    },
+  },
+];
+
+const toRegistrationWithDefaultPriority = ({
+  priority = 50,
+  ...rest
+}: WorkloadsOverviewDetailRegistration) => ({
+  priority,
+  ...rest,
+});
+
+const getRegistrationsInPriorityOrder = (
+  allRegistrations: WorkloadsOverviewDetailRegistration[],
+) =>
+  orderBy(
+    "priority",
+    "desc",
+
+    allRegistrations.map(toRegistrationWithDefaultPriority),
+  );
+
```

Comment:
Why is this done differently than done in https://github.com/lensapp/lens/pull/4820?

Reply:
Use-case  specific logic. This is just extracted to function to make code cleaner.

Reply:
We could share `orderBy("priority", "desc")` if we wanted to, but I don't see reason.

Reply:
See my response on the other PR

Reply:
Done in #4820